src/main/java/com/google/vinet/servlets/RequestServlet.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+
+import com.google.vinet.data.*;
+import java.time.*;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.stream.Stream;
+
+/**
+ * A web servlet for posting user requests.
+ */
+@WebServlet("/request")
+public class RequestServlet extends HttpServlet {
+  /** The Datastore entity name for the ticket(s) associated with a request. */
+  public static final String TICKET_TABLE_NAME = "Ticket";
+  /** The {@code DatastoreService} implementation that this {@code RequestServlet depends on}. */
+  private final DatastoreService datastore;
+  /** The {@code UserService} implementation that this {@code RequestServlet depends on}. */
+  private final UserService userService;
+  /** The {@code RegistrationServlet} implementation that this {@code RequestServlet depends on}. */
+  private final RegistrationServlet registrationServlet;
+
+  /**
+   * Construct a RequestServlet with all of its dependencies set to their default implementations.
+   */
+  public RequestServlet() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.userService = UserServiceFactory.getUserService();
+    this.registrationServlet = new RegistrationServlet();
+  }
```

Comment:
We can't access the code that initialises the servlets, since that's fully managed by App Engine / Jetty, so they have to be initialised for each servlet that's constructed. The lifetime of a servlet is very short-lived, so the connections won't exist once the response has been written to and the servlet has been destroyed.

Reply:
Ah, fair enough